DelvUI/Interface/StatusEffects/StatusEffectsListHud.cs

```diff
@@ -199,6 +206,28 @@ protected unsafe List<StatusEffectData> StatusEffectDataList(GameObject? actor)
             return list;
         }
 
+        protected bool IsStatusFromPlayerPet(StatusStruct status)
+        {
+            var player = Plugin.ClientState.LocalPlayer;
+            if (player == null)
+            {
+                return false;
+            }
+            
+            var character = Plugin.ObjectTable.SearchById(status.SourceID);
```

Comment:
Doing this every frame for every status for every list is pretty costly.
I think you can directly find the pet's address/id by using the BuddyList class from Dalamud, but haven't tested it.

Reply:
Well the only other options I thought off is including it in the StatusEffect struct but that required a whole more bunch of changes to if everywhere so wasn't sure what option would be best.

Reply:
I think the way you did it is fine if we can simplify the look up for the pet's actor.
I'm pretty sure you can get it directly with the class I mentioned above, but again, haven't really tested it.

Reply:
Actually that was exactly what I needed, thanks for bringing that class up, works for me :)